src/commonMain/kotlin/CommonJsonLogicEngine.kt

```diff
@@ -70,17 +72,20 @@ internal class CommonJsonLogicEngine : JsonLogicEngine {
 
     private fun apply(logic: Any?, data: Any?): Any? {
         return when {
+            logic is List<*> -> logic.map { apply(it, data) }
             logic !is Map<*, *> -> logic
             logic.isEmpty() -> data
-            else -> {
-                val operator = logic.keys.firstOrNull()
-                val values = logic[operator]
-                operations[operator]?.invoke(when (values) {
-                    is List<*> -> values.map { apply(it, data) }
-                    is Map<*, *> -> apply(values, data)
-                    else -> apply(listOf(values), data)
-                }.asList, data)
-            }
+            else -> execute(logic, data)
         }
     }
+
+    private fun execute(logic: Map<*, *>, data: Any?): Any? {
```

Comment:
I'm not sure this function name and `apply` name are descriptive enough. It's a pretty simple and short class so it's not a blocker  but if you could came up with some better name it would be great